public/app/features/dashboard/components/PanelEditor/OptionsPaneItemDescriptor.tsx

```diff
@@ -29,74 +31,79 @@ export interface OptionsPaneItemProps {
 export class OptionsPaneItemDescriptor {
   parent!: OptionsPaneCategoryDescriptor;
 
-  constructor(public props: OptionsPaneItemProps) {}
+  constructor(public props: OptionsPaneItemInfo) {}
 
-  getLabel(searchQuery?: string): ReactNode {
-    const { title, description, overrides, addon } = this.props;
-
-    if (!searchQuery) {
-      // Do not render label for categories with only one child
-      if (this.parent.props.title === title && !overrides?.length) {
-        return null;
-      }
-
-      return <OptionPaneLabel title={title} description={description} overrides={overrides} addon={addon} />;
-    }
-
-    const categories: React.ReactNode[] = [];
+  render(searchQuery?: string) {
+    return <OptionsPaneItem key={this.props.title} itemDescriptor={this} searchQuery={searchQuery} />;
+  }
+}
 
-    if (this.parent.parent) {
-      categories.push(this.highlightWord(this.parent.parent.props.title, searchQuery));
-    }
+interface OptionsPaneItemProps {
+  itemDescriptor: OptionsPaneItemDescriptor;
+  searchQuery?: string;
+}
 
-    if (this.parent.props.title !== title) {
-      categories.push(this.highlightWord(this.parent.props.title, searchQuery));
-    }
+function OptionsPaneItem({ itemDescriptor, searchQuery }: OptionsPaneItemProps) {
+  const { title, description, render, showIf, skipField, useShowIf } = itemDescriptor.props;
+  const key = `${itemDescriptor.parent.props.id} ${title}`;
 
-    return (
-      <Label description={description && this.highlightWord(description, searchQuery)} category={categories}>
-        {this.highlightWord(title, searchQuery)}
-        {overrides && overrides.length > 0 && <OptionsPaneItemOverrides overrides={overrides} />}
-      </Label>
-    );
+  // eslint-disable-next-line react-hooks/rules-of-hooks
+  if (useShowIf && !useShowIf()) {
+    return null;
```

Comment:
How about making useShowIf an instance method on `OptionsPaneItemDescriptor` that would return either `false` by default or configured hook? Then we wouldn't have to depend on reading descriptor's props and executing hooks conditionaly.

Reply:
@dprokop good idea! moved this to a useShowIf function on OptionsPaneItemDescriptor the nice thing with that is that that function can also check the old showif function and use that if that is defined, that way we only need one if statement in the render function